### ISSUE TITLE
Panic on resize in iced_glow-0.2.0 and wgpu-0.7.1

### ISSUE THREAD

**Issue by DusterTheFirst** [iced-rs/iced#1111]
The glow and wgpu back-ends will both panic when resizing my application vertically. Interestingly I have not noticed this happening when resizing horizontally or when expanding vertically, only when the resizing shrinks the window vertically . Since the issue persists across both back-ends, this leads me to believe that this could be a problem with the `plotters-iced` crate sending bad data to the iced back-ends. If this issue is not one with iced, I will raise the issue to the `plotters-iced` crate. I wanted to make sure that it is not a problem with `iced` itself before I raise the issue to `plotters-iced`.

Here is the code in question: https://github.com/DusterTheFirst/mission-control/tree/5fd737c99b5a55b9803ecc862428e233f6a3446a/crates/ground-station

I also want to thank the whole team for all of the work they have done on this crate to get it to how it is today. It has been one of the crates that has given me the best UI creation experience.

**EDIT (6 hours later)**: I have refactored the layout of my iced code for reasons non-related to this issue, but it seems to have removed this panic from occurring. The previous layout was an attempt at a grid layout with one row with 5 columns in it, each column containing 5 other elements to create a faux grid which you can see here: 

https://github.com/DusterTheFirst/mission-control/blob/5fd737c99b5a55b9803ecc862428e233f6a3446a/crates/ground-station/src/main.rs#L185-L279

My new layout no longer tries to mimic a grid as it was too restrictive for my liking and no longer encounters a panic when shrinking the window. The code for the changed layout can be seen here:

https://github.com/DusterTheFirst/mission-control/blob/ec9bf6d44903f4eaa59ddc0590af8ada01eb050b/crates/ground-station/src/main.rs#L178-L281

**Continuing from before the edit**

## Glow panic

https://user-images.githubusercontent.com/14093962/140608488-82d6fd61-be67-4be5-adf3-7833ea1849bb.mp4

```
thread 'main' panicked at 'attempt to subtract with overflow', C:\Users\<username>\.cargo\registry\src\github.com-1ecc6299db9ec823\iced_glow-0.2.0\src\triangle.rs:195:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## Wgpu panic

https://user-images.githubusercontent.com/14093962/140608671-fb344e10-8a34-4a65-8e91-caf33498c1e4.mp4

```
wgpu error: Validation Error

Caused by:
    In a RenderPass
      note: encoder = `iced_wgpu encoder`
    In a set_scissor_rect command
    Invalid ScissorRect parameters


thread 'main' panicked at 'Handling wgpu errors as fatal by default', C:\Users\<username>\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.1\src\backend\direct.rs:1896:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by SeanTater** [comment]
I experience the same error, using wgpu on Arch Linux with Intel + Vulkan, using the Todo example, with Iced 0.3.0 and the Todo taken from the commit tagged 0.3.0.

**Comment by rdrpenguin04** [comment]
Same error, with the `Hello` example in the `Application` docs: https://docs.rs/iced/0.3.0/iced/trait.Application.html

**Comment by hecrj** [comment]
I think this should be fixed in `master` by #818.
